File: scala/004/Main.scala
/*
 * ProjectEuler 004
 * Scala / Main.scala
 * copyright (c) 2015 Susisu
 *
 * Scala 2.11.4
 * `scalac Main.scala -d bin`
 */

import euler._

object Main {
    def main(args: Array[String]) {
        var max = 0;
        for (a <- 100 until 1000) {
            for (b <- a until 1000) {
                val p = a * b;
                if (isPalindrome(p) && p > max) {
                    max = p;
                }
            }
        }
        println(max);
    }
}

package object euler {
    def isPalindrome(n: Int): Boolean = {
        val s = n.toString();
        return s == s.reverse;
    }
}

File: java/004/Main.java
/*
 * ProjectEuler 004
 * Java / Main.java
 * copyright (c) 2015 Susisu
 *
 * Java SE 1.8.0_25
 * `javac Main.java -d bin`
 */

public class Main {
    public static void main(String[] args) {
        int max = 0;
        for (int a = 100; a < 1000; a++) {
            for (int b = a; b < 1000; b++) {
                int p = a * b;
                if (Euler.isPalindrome(p) && p > max) {
                    max = p;
                }
            }
        }
        System.out.println(max);
    }
}
